broadway/api/handlers/client.py

```diff
@@ -270,3 +270,56 @@ def get(self, *args, **kwargs):
                 {"message": "scope {} has not been implemented yet".format(scope)}, 404
             )
             return
+
+
+class CourseQueueLengthHandler(ClientAPIHandler):
+    @authenticate_course
+    @schema.validate(
+        output_schema={
+            "type": "object",
+            "properties": {"length": {"type": "number"}},
+            "required": ["length"],
+            "additionalProperties": False,
+        },
+        on_empty_404=True,
+    )
+    def get(self, *args, **kwargs):
+        course_id = kwargs["course_id"]
+        queue = self.settings["QUEUE"]
+
+        if queue.contains_key(course_id):
+            return {"length": queue.get_queue_length(course_id)}
+        else:
+            self.abort(
+                {"message": f"{course_id} does not exist as a course in the queue"}
+            )
+            return
+
+
+class GradingRunQueuePositionHandler(ClientAPIHandler):
+    @authenticate_course
+    @schema.validate(
+        output_schema={
+            "type": "object",
+            "properties": {"position": {"type": "number"}},
+            "required": ["position"],
+            "additionalProperties": False,
+        },
+        on_empty_404=True,
+    )
+    def get(self, *args, **kwargs):
+        course_id = kwargs["course_id"]
+        grading_run_id = kwargs.get("run_id")
+        queue = self.settings["QUEUE"]
+
+        if not queue.contains_key(course_id):
+            self.abort(
+                {"message": f"{course_id} does not exist as a course in the queue"}
+            )
+            return
+
+        queue_position = queue.get_position_in_queue(course_id, grading_run_id)
+        if queue_position == -1:
+            self.abort({"message": f"{grading_run_id} was not found in the queue"})
```

Comment:
We should be able to differentiate between "a grading job with this run id doesn't exist" vs "a grading job has finished" when asking for position in queue. How about add a check before these lines to see if this grading job actually exist. If it doesn't, send an error saying grading job doesn't exist. If it does, and you can't find it in the queue, it must mean that the grading job is finished (is this true?), then send a different message.